### PR TITLE
[Java][Spring] fix RequestPart/RequestParam handling on multipart req…

### LINKED ISSUE #14449: [BUG][JAVA][SPRING] Wrong annotation used for array of referenced object in Multipart request
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

When we specify the request body as `multipart/form-data` and have one property of type array.

The generated annotation for the `statusArray` property is "@RequestParam". Which leads to following error in Spring when the endpoint is used:
```
org.springframework.web.method.annotation.MethodArgumentConversionNotSupportedException:
Failed to convert value of type 'java.lang.String' to required type 'List.of<MultipartMixedStatus>';
```
Expected would be the annotation "@RequestPart" which fixes the Spring error.

##### openapi-generator version

6.2.1

But the problem was probably introduced in #11449 with version 6.0.0 ([thanks to @borsch to figure this out](https://github.com/OpenAPITools/openapi-generator/issues/14295#issuecomment-1368773203))

##### OpenAPI declaration file content or url

```yaml
[...]
requestBody:
  content:
    multipart/form-data:
      schema:
        type: object
        properties:
          file:
            description: "a file"
            type: string
            format: binary
          statusArray:
            type: array
            items:
              $ref: '#/components/schemas/MultipartMixedStatus'
[...]
```

##### Generation Details

We used the Maven plugin with following config

```xml
        <configuration>
          <templateDirectory>openapi-generator-customization</templateDirectory>
          <generatorName>spring</generatorName>
          <modelNameSuffix>Dto</modelNameSuffix>
          <configOptions>
            <useSpringBoot3>true</useSpringBoot3>
            <documentationProvider>springdoc</documentationProvider>
            <useTags>true</useTags>
            <useBeanValidation>true</useBeanValidation>
            <performBeanValidation>true</performBeanValidation>
            <useSpringController>true</useSpringController>
            <oas3>true</oas3>
            <apiPackage>com.example.api</apiPackage>
            <modelPackage>com.example.model</modelPackage>
            <invokerPackage>com.example.invoker</invokerPackage>
            <interfaceOnly>true</interfaceOnly>
            <skipOverwrite>false</skipOverwrite>
          </configOptions>
        </configuration>
```

##### Steps to reproduce

See test in PR #14450.

##### Related issues/PRs

- #14295 fixed the problem for not wrapped objects
- #11449 might have introduced the problem

##### Suggest a fix

Fixed by: #14450

### PR DESCRIPTION
**Pull request by MetroMarv**
…uest

For arrays

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@cachescrubber @welshm @MelleD @atextor @manedev79 @javisst @borsch @banlevente @Zomzog 

fix #14449

### CODE REVIEW

**Review by borsch — APPROVED** [review]

**Comment by MetroMarv** [comment]
I didn't build the whole project and generate the samples. As some of the tests fail on my local machine. Is this a local problem? Because the CircleCI tests seem to pass.
```
[ERROR] Failures: 
[ERROR]   DefaultGeneratorTest.supportCustomTemplateEngine:313 » Runtime Could not generate api file for 'Ping'
[ERROR]   DefaultGeneratorTest.testProcessUserDefinedTemplatesWithConfig:726 » Runtime Could not generate model 'ApiResponse'
[ERROR]   PythonClientTest.testApiTestsNotGenerated:143 » Runtime Could not generate model 'ApiResponse'
[ERROR]   PythonClientTest.testApisNotGenerated:164 » Runtime Could not generate model 'ApiResponse'
[ERROR]   PythonPriorClientTest.testObjectModelWithRefedAdditionalPropertiesIsGenerated:408 » Runtime Could not generate model 'A'
[ERROR]   StringHelpersTest.startsWithSectionalTest:47->evaluate:28 » Handlebars inline@486b7642:1:4: java.lang.reflect.InaccessibleObjectException: Unable to make field transient java.util.HashMap$Node[] java.util.HashMap.table accessible: module java.base does not "opens java.util" to unnamed module @43814d18
    inline@486b7642:1:4
[ERROR]   StringHelpersTest.startsWithTest:61->evaluate:28 » Handlebars inline@43bd77be:1:2: java.lang.reflect.InaccessibleObjectException: Unable to make field transient java.util.HashMap$Node[] java.util.HashMap.table accessible: module java.base does not "opens java.util" to unnamed module @43814d18
    inline@43bd77be:1:2
[ERROR]   StringHelpersTest.startsWithYesOverrideTest:81->evaluate:28 » Handlebars inline@689e2599:1:2: java.lang.reflect.InaccessibleObjectException: Unable to make field transient java.util.HashMap$Node[] java.util.HashMap.table accessible: module java.base does not "opens java.util" to unnamed module @43814d18
    inline@689e2599:1:2
[INFO] 
[ERROR] Tests run: 1850, Failures: 8, Errors: 0, Skipped: 0

```

**Comment by wing328** [comment]
> As some of the tests fail on my local machine.

are you using JDK17 ?

**Comment by MetroMarv** [comment]
Yes

` ` `
Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)
Maven home: /home/marvin/.m2/wrapper/dists/apache-maven-3.8.4-bin/52ccbt68d252mdldqsfsn03jlf/apache-maven-3.8.4
Java version: 17.0.5, vendor: N/A, runtime: /usr/lib/jvm/java-17-openjdk
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "6.1.1-1-manjaro", arch: "amd64", family: "unix"
` ` `

**Comment by wing328** [comment]
please use `mvn clean package -DskipTests -Dmaven.javadoc.skip=true` for the time being.

**Comment by MetroMarv** [comment]
Alright. Samples were generated. Apologies for the inconvenience.

**Comment by MetroMarv** [comment]
Could you

**Comment by MetroMarv** [comment]
Is there anything else that we need to do in order to get this merged?

File: modules/openapi-generator/src/main/resources/JavaSpring/formParams.mustache [hunk 1]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+{{#isFormParam}}{{^isFile}}{{>paramDoc}}{{#useBeanValidation}} @Valid{{/useBeanValidation}} {{#isModel}}@RequestPart{{/isModel}}{{^isModel}}{{#isArray}}@RequestPart{{/isArray}}{{^isArray}}@RequestParam{{/isArray}}{{/isModel}}(value = "{{baseName}}"{{#required}}, required = true{{/required}}{{^required}}, required = false{{/required}}){{>dateTimeParam}} {{{dataType}}} {{paramName}}{{/isFile}}{{#isFile}}{{>paramDoc}} @RequestPart(value = "{{baseName}}"{{#required}}, required = true{{/required}}{{^required}}, required = false{{/required}}) {{#isArray}}List<{{/isArray}}{{#reactive}}Flux<Part>{{/reactive}}{{^reactive}}MultipartFile{{/reactive}}{{#isArray}}>{{/isArray}} {{paramName}}{{/isFile}}{{/isFormParam}}
```
Comment (borsch) [inline modules/openapi-generator/src/main/resources/JavaSpring/formParams.mustache:1]: could you please new line at the end? too many samples are unnecessary updated

**Review by borsch — CHANGES_REQUESTED** [review]

File: modules/openapi-generator/src/main/resources/JavaSpring/formParams.mustache (hunk 1 above)
Comment (MetroMarv) [reply modules/openapi-generator/src/main/resources/JavaSpring/formParams.mustache]: Fixed.

Comment (borsch) [reply modules/openapi-generator/src/main/resources/JavaSpring/formParams.mustache]: thanks

**Review by borsch — APPROVED** [review]

**Comment by borsch** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by MetroMarv** [comment]
Thanks for pointing that out. Changed the commit's author's email now :).

**Comment by borsch** [comment]
@MetroMarv on "Commits" tab and commit it self seems like you account doesn't referenced yet

**Comment by MetroMarv** [comment]
Could you please check again?

**Pull request merged**